### ISSUE TITLE
Implement compilation and execution of new test syntax

### ISSUE THREAD

**Issue by tstirrat15** [authzed/spicedb#2134]
Part of #2127 

Once the parser is implemented, we'll need to use those AST nodes. My best guess of how this will work:

1. Implement translator logic to turn those nodes into internal representations
2. Construct a dev context using those representations and attach it to the compiled schema
3. Add logic to execute the validations attached to the compiled schema

I think there's probably a conversation to be had about when/where the validations are to be run.

**Comment by vroldanbet** [comment]
@tstirrat15 could you provide a bit more context for this?

**Comment by tstirrat15** [comment]
Yeah, this was a tracking issue for inline testing syntax from when we thought that was going to be a part of composable schema syntax. We can probably close this for now.
